community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/IndexTransactionApplier.java

```diff
@@ -103,15 +104,42 @@ private void updateIndexes() throws IOException, IndexCapacityExceededException,
 
     private void updateLabelScanStore() throws IOException, IndexCapacityExceededException
     {
-        // Even if the node commands are sorted by node id, they are not totally ordered by node id,
-        // since created records comes before updated records, which comes before deleted records.
-        Collections.sort( labelUpdates, SORT_BY_NODE_ID );
+        // Updates are sorted according to node id here, an artifact of node commands being sorted
+        // by node id when extracting from TransactionRecordState.
+        labelScanStoreSync.apply(
+                new LabelUpdateWork( new ArrayList<>( labelUpdates ) ) );
```

Comment:
You're doing a safe copy of `labelUpdates` to guarantee it being mutable, or?. Since `LabelUpdateWork` is internal to `IndexTransactionApplier` I think it's safe to not doing a copy, knowing that `labelUpdates` is already mutable.